### PR TITLE
Add DuckDB runner and fix-up assorted queries

### PR DESCRIPTION
**Pull request by gforsyth**
Breaking this out from the work in #3 as that will probably take a while to finish.

Adds a DuckDB Runner, fixes a few issues that showed up when running with DuckDB (like the case of certain string arguments) and fixes the incorrect comparison sign in query one (it's a `<=` not a `<`)

### CODE REVIEW

File: ibis_tpc/runners.py [hunk 1]
```diff
@@ -236,13 +255,14 @@ def run(self, qid, outdir=None, backend="sqlite"):
...
 setup_dplyr = RRunner
 setup_dbplyr = RRunner
 
 
 def compare(rows1, rows2):
     diffs = []
-    for i, (r1, r2) in enumerate(itertools.zip_longest(rows1, rows2)):
+    for row, (r1, r2) in enumerate(itertools.zip_longest(rows1, rows2)):
         if r1 is None:
             diffs.append(f"[{i}]  extra row: {r2}")
```
Comment (saulpw) [inline ibis_tpc/runners.py]: `{i}` should be `{row}`; could we also rename `row` to `rownum`?

**Comment by gforsyth** [comment]
I think this is more or less set -- just waiting on a new release of Ibis and then I can squash and update the dependencies (and remove the `git clone` bits)

**Comment by gforsyth** [comment]
@saulpw -- this is ready for another look now that ibis 3.0.0 is available

**Review by saulpw — APPROVED** [review]

**Comment by saulpw** [comment]
Thanks @gforsyth for putting this together!

**Pull request merged**